go-controller/pkg/ovn/egressgw.go

```diff
@@ -616,11 +616,10 @@ func (oc *Controller) cleanECMPRoutes() {
 		node := util.GetWorkerFromGatewayRouter(gr)
 		prefix, err := oc.extSwitchPrefix(node)
 		if err != nil {
-			klog.Errorf("cleanECMPRoutes: failed to find logical router for %s", uuid, err, stderr)
+			klog.Errorf("cleanECMPRoutes: failed to find logical router for %s, err: %v", uuid, err)
 			continue
 		}
-		if (prefix != "" && !strings.Contains(port, prefix)) ||
-			(prefix == "" && strings.Contains(port, prefix)) {
```

Comment:
To reviewers: this is the bug. And empty string will always match here, hence the condition is evaluated as: 

```
if (prefix != "" && !strings.Contains(port, prefix)) || true {
```

or if you will 

```
if true {
```

https://play.golang.org/p/hOmAqeZi_a8

Reply:
@alexanderConstantinescu : see https://github.com/ovn-org/ovn-kubernetes/pull/2348

Reply:
Doh! So there were a lot more bugs than this it seems. Closing this PR.